hudi-spark-datasource/hudi-spark3.2plus-common/src/main/scala/org/apache/spark/sql/catalyst/plans/logcal/HoodieTableChanges.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans.logcal
+
+import org.apache.hudi.common.util.ValidationUtils.checkState
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.catalyst.plans.logical.LeafNode
+
+object HoodieTableChangesOptionsParser {
+  def parseOptions(exprs: Seq[Expression], funcName: String): (String, Map[String, String]) = {
+    val args = exprs.map(_.eval().toString)
+
+    if (args.size < 3) {
+      throw new AnalysisException(s"Too few arguments for function `$funcName`")
```

Comment:
Glance over the changes quickly and have some concerns:

1. where does the name like `hudi_table_changes` and `hudi_table_changes_by_path` come from, like they both work for incremental queries
2. if we can impl the named param for the table value functions, we can merge the two functions into one, for named param, it looks like: `table=xxx, ...` or `path=xxx, ...`, and we can even enable/disable the cdc loggins with the param

Reply:
AFAIK TVF in Spark unlike procedures doesn't support named arguments natively.
Looking at Spark docs: https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-tvf.html there are none that have named arguments.

So to handle everything in one TVF it would look like this:
```
hudi_table_changes('path', 'my/table', 'latest_state', '12345678')
hudi_table_changes('table', 'myTable', 'cdc', '12345678')
```
Or I would need to work against what Spark offers and parse named arguments myself to have:
```
hudi_table_changes('path=my/table', 'type=latest_state', 'beginInstantTime=12345678')
hudi_table_changes('table=myTable', 'type=cdc', 'beginInstantTime=12345678')
```
This does not feel like the right thing to do.
I wanted it to be as easy to use as it can be, just pass arguments.
In the case of path and table I can't assume what will be passed by the user and it would be hard to infer this from the input, but latest_state or cdc I can so I could do it easily and introduce this flag.

To be honest I was thinking in the other direction, to split cdc and latest_state to separate functions:
```
hudi_table_changes('table', 'earliest', '123456789')
hudi_table_changes_by_path('my/table', 'earliest', '123456789')
hudi_table_cdc_changes('table', 'earliest', '123456789')
hudi_table_cdc_changes_by_path('my/table', 'earliest', '123456789')
```

Reply:
> This does not feel like the right thing to do.
I wanted it to be as easy to use as it can be, just pass arguments

The SQL standard in 2017 definds that the table valued function could take named param, it deserves a try here.

Reply:
I only found that in one case in pgsql dblink function uses 
```sql
select * from
dblink('dbname=mydb options=-csearch_path=', 'select proname, prosrc from pg_proc')
```
But that's a db connection string passed as one argument so not exactly the same case.

I've only seen tvf arguments to be resolved by position. The declaration of func arguments can be done by name, but I have not seen anywhere that I can call my function by using `select * from myfunc(path='abc')`.

For spark I found one ticket that aims to implement named argument support in parser for function call:
https://issues.apache.org/jira/browse/SPARK-43922
targeted for Spark 3.5.

Tbh I'm not fully convinced if Spark SQL syntax does not support func call with named parameters.
This is how another implementation for similar function in spark:
https://learn.microsoft.com/en-us/azure/databricks/delta/delta-change-data-feed#sql

But what should the syntax in the mentioned standard look like then? I was not able to find any information on what the sql standard defines.
Can you give me an example? Should only the first argument be either "path=xxx/xxx", "identifier=db.tableName", or "identifier=tableName" and the rest stay as is so that users don't need to write too much in their queries?

Reply:
Ok i found this https://www.postgresql.org/docs/current/sql-syntax-calling-funcs.html#SQL-SYNTAX-CALLING-FUNCS-NAMED

Still similar syntax is not supported in Spark SQL. I think it will be nice to use it, but only after there's support for it in spark.

Wdyt Danny?

Reply:
[c069776_ISO_IEC_TR_19075-7_2017.pdf](https://github.com/apache/hudi/files/11659369/c069776_ISO_IEC_TR_19075-7_2017.pdf)
Here is the part about table valued functions from SQL standard 2017, is there any possibility we can add a custom syntax parse block for Hudi then?

Reply:
Thanks! I'll check and give it a try